src/hotspot/cpu/x86/interp_masm_x86.cpp

```diff
@@ -54,15 +54,28 @@ void InterpreterMacroAssembler::jump_to_entry(address entry) {
 void InterpreterMacroAssembler::profile_obj_type(Register obj, const Address& mdo_addr) {
   Label update, next, none;
 
+#ifdef _LP64
+  assert_different_registers(obj, rscratch1, mdo_addr.base(), mdo_addr.index());
```

Comment:
Same here: pass `rscratch1` unconditionally?

Reply:
Unfortunately, I don't know if mdo_addr.index() is a register.  I wish we had a smarter assert_different_registers() that would ignore multiple `noreg`.

Reply:
Ok, I see. That's unfortunate. IMO it's perfectly fine to ignore `noreg` parameters in `assert_different_registers`. (After all, `noreg` is mostly unusable, so it doesn't matter whether there are multiple `noreg`s or just a single one passed.) But such a cleanup can be done later.